lib/solveEquation/stepThrough.js

```diff
@@ -158,64 +126,110 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
+                      ? flattenSolutionsList.push(...s.items)
+                      : flattenSolutionsList.push(s));
+    allSolutions =  Node.Creator.list(flattenSolutionsList);
+  }
+  else if (solutions.length === 1) {
+    allSolutions = solutions[0];
+  }
+  else {
+    allSolutions = Node.Creator.list([]);
+  }
+
+  const roots = new Equation(symbol, allSolutions, '=');
+
+  return new EquationStatus(ChangeTypes.FIND_ROOTS, equation, roots);
+}
+
+/*
   Helper function that returns the roots and symbol of an input equation
-  that has roots to be found.
+  that has roots to be found. (We check if the equation has roots first)
   For every factor on the left hand side, solve a new equation that is factor = 0
   for the symbol.
   TODO: handle multiple variable solutions e.g (x + 2) (y + 2) = 0
 */
 
 function getSolutionsAndSymbol (equation) {
-  const left = equation.leftNode;
-  const rightNode = Node.Creator.constant(0);
+  const leftNode = equation.leftNode;
 
   const solutions = [];
 
   let symbol, steps, factorsWithSymbols;
 
-  // If left hand side is a power node and it does not resolve to a constant (e.g. 2^7)
-  // then it is a factor. If left hand side is a multiplication node, return a list of all the valid factors.
-  if (Node.Type.isOperator(left, '^') && !checks.resolvesToConstant(left)){
-    factorsWithSymbols = [left];
+  // If left hand side is a power node and it does not resolve to a constant
+  // then it is a factor. (e.g. 2^7 resolves to a constant so it is not a factor, but
+  // x^2 would have factors x = 0)
+  // If left hand side is a multiplication node, return a list of all the valid factors.
+  if (Node.Type.isOperator(leftNode, '^') && !checks.resolvesToConstant(leftNode)){
+    factorsWithSymbols = [leftNode];
   }
   else {
     factorsWithSymbols = equation.leftNode.args.filter(arg => !checks.resolvesToConstant(arg));
   }
 
-  // For each factor, solve the equation factor = 0 for the symbol
-  // If the factor is a power node, solve the equation base = 0 for the symbol
-  for (var f in factorsWithSymbols) {
-    let factor = factorsWithSymbols[f];
+  /*
+    For each factor, solve the equation factor = 0 for the symbol
+    If the factor is a power node, solve the equation base = 0 for the symbol
+    Show duplicate roots when factor is a power node
+    e.g. (x + 1)^2 -> x = [-1, -1]
+  */
+  try {
+    for (var f in factorsWithSymbols) {
+      let factor = factorsWithSymbols[f];
+      let exponent = 1;
+
+      if (Node.Type.isOperator(factor, '^')) {
+        exponent = parseFloat(factor.args[1].value);
+        factor = factor.args[0];
+      }
 
-    if (Node.Type.isOperator(factor, '^')) {
-      factor = factor.args[0];
-    }
+      const leftNode = Node.Type.isParenthesis(factor)
+            ? factor.content
+            : factor;
 
-    const leftNode = Node.Type.isParenthesis(factor)
-      ? factor.content
-      : factor;
+      const rightNode = Node.Creator.constant(0);
```

Comment:
this will always be `equation.rightNode`, I think it's more clear to just do

`steps = stepThrough(leftNode, equation.rightNode, '=');`